addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -189,14 +238,18 @@ def _is_compatible_with_journal(self, journal):
         self.ensure_one()
         if self.code != 'fattura_pa':
             return super()._is_compatible_with_journal(journal)
-        return journal.type == 'sale' and journal.country_code == 'IT'
+        return journal.type in ('sale', 'purchase') and journal.country_code == 'IT'
 
     def _l10n_it_edi_is_required_for_invoice(self, invoice):
         """ Is the edi required for this invoice based on the method (here: PEC mail)
             Deprecated: in future release PEC mail will be removed.
             TO OVERRIDE
         """
-        return invoice.is_sale_document() and invoice.l10n_it_send_state not in ('sent', 'delivered', 'delivered_accepted') and invoice.country_code == 'IT'
+        return all([
+            invoice.country_code == 'IT',
+            invoice.l10n_it_send_state not in ('sent', 'delivered', 'delivered_accepted'),
+            self._l10n_it_get_document_type(invoice),
```

Comment:
It would not hurt to make the distinction between sale and purchase before.  As for sale, you always need it and not always for purchase.

Reply:
Memo Just leave the "if it's a sale then send it" condition before